src/common/memory_desc.hpp

```diff
@@ -23,6 +23,8 @@
 namespace dnnl {
 namespace impl {
 
+enum class cublaslt_memory_format_t { ampere_blocked };
```

Comment:
This might be confusing since on ampere, two different blocked layouts are used for IMMA depending on which matrix it refers to. From [cublasLt documentation](https://docs.nvidia.com/cuda/cublas/#cublasltmatmul):
```
Using the IMMA-specific data ordering on Ampere or Turing (but not Hopper) architecture - CUBLASLT_ORDER_COL32` for matrices A, C, D, and CUBLASLT_ORDER_COL4_4R2_8C (on Turing or Ampere architecture) or CUBLASLT_ORDER_COL32_2R_4R4 (on Ampere architecture) for matrix B
```

I wonder how the reorder primitive can here distinguish between CUBLASLT_ORDER_COL4_4R2_8C and CUBLASLT_ORDER_COL32_2R_4R4.

It seems we might need to use the same tags as [cublasLt documents](https://docs.nvidia.com/cuda/cublas/#cublasltorder-t), e.g. `col4_4r2_8c` to map to `CUBLASLT_ORDER_COL4_4R2_8C` and `col32_2r_4r4` to map to `CUBLASLT_ORDER_COL32_2R_4R4` ?

Reply:
As discussed offline, it is perfectly fine to support CUBLASLT_ORDER_COL32_2R_4R4 only, but it is preferable to map the cublas tags here to simplify future extensions (as mapping to arch is not a one-to-one mapping).

Reply:
Changed discussed have been made, currently containing col32_2r_4r4. With future work to extend for additional format.